kornia/morphology/morphology.py

```diff
@@ -563,3 +566,52 @@ def bottom_hat(
         )
         - tensor
     )
+
+
+def skeletonize(data: Tensor, kernel: Tensor | None = None) -> Tensor:
+    r"""Return the skeleton of a binary image.
+
+    In morphological terms, it involves reducing foreground regions in a
+    binary image while preserving the extent and connectivity of the original
+    region and throwing away most of the foreground pixels.
+
+    The kernel must have 2 dimensions.
+
+    Args:
+        data: Image with shape :math:`(B, 1, H, W)`. [C = 1]
+        kernel: Positions of non-infinite elements of a flat structuring element.
+                Non-zero values give the set of neighbors of the center over which
+                the operation is applied. Its shape is :math:`(k_x, k_y)`.
+                For full structural elements use torch.ones_like(structural_element).
+    Returns:
+        Skeleton of the image with shape :math:`(B, 1, H, W)`. [C = 1]
+
+    Example:
+        >>> data = torch.rand(1, 1, 20, 20)
+        >>> kernel = torch.ones(3, 3)
+        >>> skeleton_img = skeletonize(data, kernel)
+    """
+
+    KORNIA_CHECK_IS_TENSOR(data)
+    KORNIA_CHECK(len(data.shape) == 4, "Input size must have 4 dimensions.")
+
+    data = data.int()
+    output = torch.zeros(data.shape, device=data.device, dtype=torch.int64)
+
+    if kernel is None:
+        kernel = torch.tensor([[0, 1, 0], [1, 1, 1], [0, 1, 0]], device=data.device, dtype=torch.int64)
+
+    KORNIA_CHECK_IS_TENSOR(kernel)
+    KORNIA_CHECK(len(kernel.shape) == 2, "Kernel size must have 2 dimensions.")
+
+    while True:
+        opened_img = opening(data, kernel)
+        subtracted_img = torch.sub(data, opened_img)
+        subtracted_img[subtracted_img < 0] = 0
+        eroded_img = erosion(data, kernel)
+        output = torch.bitwise_or(output, subtracted_img)
+        data = eroded_img.clone()
+        if torch.count_nonzero(data) == 0:
```

Comment:
how this behaves in case at least one sample of the batch hit this condition? this method looks like a candidate to work only for single images (HxW) and extend the batched version with for loop or `vmap` @johnnv1 ?

Reply:
this is true, this shouldn't work for a batched case

Reply:
i'll merge this one, just update the docs about it, and then we open as issues the remaining items